### ISSUE TITLE
PHP 7.4 Support arrow function return types

### ISSUE THREAD

**Issue by semalley** [phpmd/phpmd#731]
- PHPMD version: 7.4.2
- PHP Version: 7.4.2
- Package version: 2.8.1
- Installation type: composer
- Operating System / Distribution & Version: Linux 4.19.76

### Current Behavior
- [x] `Unexpected token:  array` for such field
`protected array $field = [];`
- [ ] And unsupported return type in arrow function
`$t = array_map(fn(int $item): int => $item * 2, [1, 2, 3]);`

### Expected Behavior
Expected success run for md command for such code
```php
class Foo
{
    public ?int $num;
    private string $str;

    public function __construct()
    {
        array_map(fn(int $item): int => $item * 2, [1, 2, 3]);
    }
}

```

### Steps To Reproduce:
Use typed fields
Use optional typed fields - `private ?string $test;`
Use return type hint in arrow function

**Comment by kylekatarnls** [comment]
Please try to require `"pdepend/pdepend": "dev-master"`

The next version will include this fix:
https://github.com/pdepend/pdepend/pull/455

**Comment by semalley** [comment]
@kylekatarnls Thanks it works for me.

**Comment by R4c00n** [comment]
Hello @kylekatarnls, I'm also facing this issue, even with your suggestion.
I have phpmd version `^2.8.2` and latest `dev-master` of pdepend installed (commit `1aa2302`).

The following code fails with `Unexpected token: :, line: 6, col: 41, file: \/Users\/r4c00n\/phpmdbug.php.`
```
<?php
declare(strict_types=1);

class Demo {
    public function foo(): array {
        return array_filter(fn(int $num): bool => $num > 5, [1, 2, 3, 4, 5, 6]);
    }
}
```

My phpmd.xml looks like this
```
<?xml version="1.0"?>
<ruleset name="Demo Ruleset"
         xmlns="http://pmd.sf.net/ruleset/1.0.0"
         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
         xsi:schemaLocation="http://pmd.sf.net/ruleset/1.0.0
                     http://pmd.sf.net/ruleset_xml_schema.xsd"
         xsi:noNamespaceSchemaLocation="
                     http://pmd.sf.net/ruleset_xml_schema.xsd">
    <description>
        PhpMd ruleset for bug demo
    </description>

    <rule ref="rulesets/cleancode.xml">
        <exclude name="StaticAccess"/>
    </rule>
    <rule ref="rulesets/codesize.xml"/>
    <rule ref="rulesets/unusedcode.xml"/>
    <rule ref="rulesets/controversial.xml/Superglobals"/>
    <rule ref="rulesets/design.xml"/>
</ruleset>

```

**Comment by kylekatarnls** [comment]
Line 6 in your example is the arrow function.

Your example is not about typed fields but about arrow function return type. Could you open a dedicated issue for that please?

Thank you.

**Comment by kylekatarnls** [comment]
I see the original issue also mentioned arrow functions. So I will rather change the title of the issue.

**Comment by R4c00n** [comment]
Yes, I know. The original issue here also reports this "And unsupported return type in arrow function"
But I can also open a new issue, no problem :)

**Comment by kevin-emo** [comment]
Hey, could it be possible to bump the current pdepend version from `2.7.1` to `2.8.0` and release a new version of phpmd in order to fix this bug which has been present since PHP 7.4 came out, please :) ?
